### PR TITLE
Fix Links don't work without a subdomain, Fixes #71112

### LINKED ISSUE #71112: Links don't work without a subdomain
Many links in VS Code do not open in extension pages and in the terminal. The behavior I see is that links without a subdomain `e.g. https://github.com` will not work, while links with a subdomain `e.g. https://www.google.com` will. 

To repro: 
1. Open terminal
2. `echo https://google.com`
3. Ctrl/Alt+Click the link

🐛 => Nothing happens.

@bpasero or @jrieken ?

### PR DESCRIPTION
**Pull request by dig22**
Added Code to check for https and http URL so it doesn't mistake it as extension id when a parent domain url is used.

`if (!isExtensionId(uri.authority) || uri.scheme === `https` || uri.scheme === `http`) {
			return false;
}`

It's my first attempt at open source contribution. Please let me know if anything needs to be changed. 
Thank you for your time. 

@joaomoreno @bpasero @jrieken @sbatten #71112

### CODE REVIEW

File: src/vs/editor/browser/services/openerService.ts [hunk 1]
```diff
@@ -43,6 +43,7 @@ export class OpenerService implements IOpenerService {
 			const handled = await opener.open(resource, options);
 			if (handled) {
 				return true;
+
```
Comment (jrieken) [inline src/vs/editor/browser/services/openerService.ts]: unnecessary change

File: src/vs/platform/opener/common/opener.ts [hunk 2]
```diff
@@ -32,5 +32,7 @@ export interface IOpenerService {
 export const NullOpenerService: IOpenerService = Object.freeze({
 	_serviceBrand: undefined,
 	registerOpener() { return { dispose() { } }; },
-	open() { return Promise.resolve(false); }
+	open() {
+		return Promise.resolve(false);
+	}
```
Comment (jrieken) [inline src/vs/platform/opener/common/opener.ts]: unnecessary change

**Review by jrieken — COMMENTED** [review]
please keep PR changes minimal

**Comment by jrieken** [comment]
Adding @joaomoreno for the actual change/fix. That code isn't known to me.

File: src/vs/platform/opener/common/opener.ts (hunk 2 above)
Comment (dig22) [reply src/vs/platform/opener/common/opener.ts]: I'm sorry i was debugging a bit, I have removed the unnecessary formatting!

**Comment by jrieken** [comment]
Tho, having said [that](https://github.com/Microsoft/vscode/pull/71218#issuecomment-477093329) I think the fix is too specific and instead the `handleUnhandledURL`-function should be fixed

**Comment by dig22** [comment]
> handleUnhandledURL

Okay I'm working on it,
Thank you

**Comment by jrieken** [comment]
Wait maybe not... The scheme-check might be way to go, but don't hard code but compare the a 'right' scheme, not a list of wrong schemes

**Comment by dig22** [comment]
> Wait maybe not... The scheme-check might be way to go, but don't hard code but compare with the 'right' scheme, not a list of wrong schemes

Okay so function to check scheme for the extensions, I'm not sure how the right scheme looks like.
Could you point me to sample extension schemes?

**Comment by dig22** [comment]
Or do you mean 
Create a function like` isHyperTextLink()`

**Comment by jrieken** [comment]
@joaomoreno know how but this specific handler only works for schemes that are the product name, e.g. `vscode`, `vscode-insider`, `vscode-oss` etc etc. That information could (should?) be in the url service itself so that the handler is invoked at all

**Review by joaomoreno — COMMENTED** [review]
The right fix for #71112 is https://github.com/Microsoft/vscode/commit/5663d6515a56b8ff0cbe172c4cd30548f3519261. We shouldn't let URL's that are meant to be opened go into the URLService world, unless they match the application's protocol scheme. We should probably rename URLService to ApplicationProtocolServer or something more OS specific.

Thanks for the attempt, @dig22, you were close!
